src/test/java/uk/nhs/prm/deduction/e2e/utility/HealthCheck.java

```diff
@@ -0,0 +1,38 @@
+package uk.nhs.prm.deduction.e2e.utility;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+
+
+public class HealthCheck {
+    private static final RestTemplate restTemplate = new RestTemplate();
+    private static final Logger LOGGER = LogManager.getLogger(HealthCheck.class);
+
+    public static boolean isHealthCheckPassing(String rootUrl) {
+        String healthCheckUrl = rootUrl + "health";
+        LOGGER.info("Checking health status of service: {}" , rootUrl);
+
+        try {
+            return await().atMost(5, TimeUnit.SECONDS)
+                    .with()
+                    .pollInterval(200, TimeUnit.MILLISECONDS)
+                    .until(() -> getRequest(healthCheckUrl).getStatusCode().is2xxSuccessful(), is(true));
+        } catch (Exception e) {
```

Comment:
Can we please change this to `exception`?

```suggestion
        } catch (Exception exception) {
```

Reply:
Thanks! Addressed this comment

Reply:
Sorry @joefong-nhs - I was meaning to add here, can we be specific to what exception we can expect to be caught here? A tip from me is if you hold your cursor over any logic, usually it'll display a definition where it will give you a lovely breakdown of the exceptions!

For example,

`try {} catch(IllegalArgumentException exception) {}`

In Java, we try to be as explicit as possible! 😄

Reply:
Thanks! I have changed it like this:
`catch (HttpClientErrorException | ConditionTimeoutException exception)`